wire/msgtx.go

```diff
@@ -309,6 +309,23 @@ func (t TxWitness) SerializeSize() int {
 	return n
 }
 
+// ToHexStrings formats the passed witness stack as a slice of hex-encoded
+// strings to be used in a JSON response.
+func (t TxWitness) ToHexStrings() []string {
+	// Ensure nil is returned when there are no entries versus an empty
+	// slice so it can properly be omitted as necessary.
+	if len(t) == 0 {
+		return nil
+	}
+
+	result := make([]string, 0, len(t))
+	for _, wit := range t {
+		result = append(result, hex.EncodeToString(wit))
```

Comment:
Whenever we have an index based loop (e.g. not iterating over a map), I think we should use the index based access:
```
	result := make([]string, len(t))
	for idx, wit := range t {
		result[idx] = hex.EncodeToString(wit)
	}
```

IMO the `append()` based version always needs a second look when reading the code, because the initial allocation can cause unwanted behavior if the middle parameter is missing (`make([]T, 0, n)` vs. `make([]T, n)`).
The index based access just makes it more clear in my eyes.